docs/references/listener.rst

```diff
@@ -0,0 +1,45 @@
+.. _listener:
+
+Listener
+########
+
+PostgREST enables a `listener to a notification channel <https://www.postgresql.org/docs/current/sql-listen.html>`_ at startup.
+It's used to :ref:`reload the Schema Cache <schema_reloading_notify>` and the :ref:`Configuration <config_reloading_notify>` from within the database via the `NOTIFY command <https://www.postgresql.org/docs/current/sql-notify.html>`_.
+
+.. code:: postgresql
+
+  NOTIFY pgrst;
+
+The ``pgrst`` notification channel is enabled by default.
+See :ref:`db-channel` and :ref:`db-channel-enabled` to configure this channel.
+
+This is useful in environments where you can’t send SIGUSR1 or SIGUSR2 Unix Signals.
+Like on cloud managed containers or on Windows systems.
+
+Listener on Read Replicas
+=========================
+
+The ``LISTEN`` and ``NOTIFY`` commands do not work on PostgreSQL read replicas.
+
+.. code:: postgresql
+
+  NOTIFY pgrst;
+
+.. code::
+
+  ERROR:  cannot execute NOTIFY during recovery
+
+If PostgREST connects to a read replica, the notification channel will fail to start.
+To prevent this, you need to provide `two hosts <https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-MULTIPLE-HOSTS>`_ in your :ref:`connection string <db-uri>`.
+The first one would be the read replica and the second one the primary database.
+For example:
+
+.. code:: bash
+
+  db-uri = "postgres://read_replica:5433,primary:5432/mydb"
+
+This way, PostgREST will handle requests in the ``read_replica`` host and listen to notifications in the fallback ``primary`` host.
+
+.. note::
+
+  Under the hood, PostgREST adds the parameter `target_session_attrs=read-write <https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNECT-TARGET-SESSION-ATTRS>`_ to the connection string when creating a listener session in the DB.
```

Comment:
```suggestion
To workaround this, you can connect the Listener to a primary database while still using the :ref:`connection_pool` on the replica.

To do this, use the standard `libpq multiple hosts <https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-MULTIPLE-HOSTS>`_ and `target_session_attrs <https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNECT-TARGET-SESSION-ATTRS>`_ in your :ref:`connection string <db-uri>`.

.. code:: bash

  db-uri = "postgres://read_replica.host,primary.host/mydb?target_session_attrs=read-only"

This will cause the :ref:`connection_pool` to connect to the read replica host and LISTEN on the fallback primary host.

.. note::

  Under the hood, PostgREST forces `target_session_attrs=read-write <https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNECT-TARGET-SESSION-ATTRS>`_ for the LISTEN session.
```